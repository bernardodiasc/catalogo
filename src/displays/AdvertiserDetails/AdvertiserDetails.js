import React, { PureComponent } from 'react'
import marked from 'marked'
import renderHTML from 'react-render-html'
import './AdvertiserDetails.css'

class AdvertiserDetails extends PureComponent {
  static defaultProps = {
    advertiser: {
      'index.md': {
        attr: {
          name: '',
          category: '',
          contact: {},
          location: {},
        },
        body: '',
      }
    },
  }

  render() {
    const { attr, body } = this.props.advertiser['index.md']
    return (
      <div className="AdvertiserDetails">
        <h1>{attr.name}</h1>
        <p>{attr.category}</p>
        {body && renderHTML(marked(body))}
        <hr />
        <ul>
          {Object.keys(attr.contact).map(each => (
            <li key={`contact-${each}`}>
              <strong>{each}:</strong>
              <span>{attr.contact[each]}</span>
            </li>
          ))}
        </ul>
        <hr />
        <ul>
          {Object.keys(attr.location).map(each => each !== 'coordinates' && (
            <li key={`location-${each}`}>
              <strong>{each}:</strong>
              <span>{attr.location[each]}</span>
            </li>
          ))}
        </ul>
      </div>
    )
  }
}

export default AdvertiserDetails
