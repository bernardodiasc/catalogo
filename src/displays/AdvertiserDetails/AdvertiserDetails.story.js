import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import Docs from 'stories/containers/Docs'

import AdvertiserDetails from './AdvertiserDetails'
import AdvertiserDetailsReadme from './README.md'
storiesOf('Displays/AdvertiserDetails', module)
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{AdvertiserDetailsReadme}</Docs>)
  .add('Example', () => <AdvertiserDetails />)
  .add('Minimal render', () => <AdvertiserDetails />)
