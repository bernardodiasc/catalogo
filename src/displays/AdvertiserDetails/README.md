# AdvertiserDetails

## How to use

```
import AdvertiserDetails from 'displays/AdvertiserDetails'
```

```
<AdvertiserDetails />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
