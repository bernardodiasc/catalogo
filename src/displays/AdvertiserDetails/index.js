import AdvertiserDetails from './AdvertiserDetails'
export default AdvertiserDetails
