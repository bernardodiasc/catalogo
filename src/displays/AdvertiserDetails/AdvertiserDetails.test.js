import React from 'react'
import { shallow } from 'enzyme'

import AdvertiserDetails from './AdvertiserDetails.js'

it('renders without crashing', () => {
  const component = shallow(<AdvertiserDetails />)
  expect(component).toMatchSnapshot()
})
