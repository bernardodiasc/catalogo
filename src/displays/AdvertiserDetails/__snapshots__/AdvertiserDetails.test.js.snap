// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="AdvertiserDetails"
>
  <h1 />
  <p />
  <hr />
  <ul />
  <hr />
  <ul />
</div>
`;
